cirq/ops/common_channels.py

```diff
@@ -50,90 +60,156 @@ def __init__(self, p_x: float, p_y: float, p_z: float) -> None:
 
     def _has_mixture_(self) -> bool:
         return True
 
     def _value_equality_values_(self):
-        return self._p_x, self._p_y, self._p_z
+        return self._num_qubits, hash(tuple(sorted(self._pis.items())))
 
     def __repr__(self) -> str:
-        return 'cirq.asymmetric_depolarize(p_x={!r},p_y={!r},p_z={!r})'.format(
-            self._p_x, self._p_y, self._p_z
-        )
+        return (
+            'cirq.asymmetric_depolarize' +
+            '(num_qubits={!r},pis={!r})'.format(self._num_qubits, self._pis))
 
     def __str__(self) -> str:
-        return 'asymmetric_depolarize(p_x={!r},p_y={!r},p_z={!r})'.format(
-            self._p_x, self._p_y, self._p_z
-        )
+        return 'asymmetric_depolarize(num_qubits={!r},pis={!r})'.format(
+            self._num_qubits, self._pis)
 
     def _circuit_diagram_info_(self,
                                args: 'protocols.CircuitDiagramInfoArgs') -> str:
-        if args.precision is not None:
-            f = '{:.' + str(args.precision) + 'g}'
-            return 'A({},{},{})'.format(f, f, f).format(self._p_x, self._p_y,
-                                                        self._p_z)
-        return 'A({!r},{!r},{!r})'.format(self._p_x, self._p_y, self._p_z)
+        if self._num_qubits == 1:
+            if args.precision is not None:
+                f = '{:.' + str(args.precision) + 'g}'
+                return 'A({},{},{})'.format(f, f,
+                                            f).format(self.p_x, self.p_y,
+                                                      self.p_z)
+            return 'A({!r},{!r},{!r})'.format(self.p_x, self.p_y, self.p_z)
+        else:
+            if args.precision is not None:
+                f = '{:.' + str(args.precision) + 'g}'
+                ps = [
+                    pauli + ':' + '{}'.format(f).format(p)
+                    for pauli, p in self._pis.items()
+                ]
+            else:
+                ps = [
+                    pauli + ':' + '{!r}'.format(p)
+                    for pauli, p in self._pis.items()
+                ]
+            return 'A({})'.format(', '.join(ps))
 
     @property
     def p_x(self) -> float:
         """The probability that a Pauli X and no other gate occurs."""
-        return self._p_x
+        assert self._num_qubits == 1
+        return self._pis.get('X', 0.0)
 
     @property
     def p_y(self) -> float:
         """The probability that a Pauli Y and no other gate occurs."""
-        return self._p_y
+        assert self._num_qubits == 1
+        return self._pis.get('Y', 0.0)
 
     @property
     def p_z(self) -> float:
         """The probability that a Pauli Z and no other gate occurs."""
-        return self._p_z
+        assert self._num_qubits == 1
+        return self._pis.get('Z', 0.0)
+
+    @property
+    def num_qubits(self) -> int:
+        """The number of qubits"""
+        return self._num_qubits
+
+    @property
+    def pis(self) -> Dict[str, float]:
```

Comment:
Let's call this "error_probabilities". This is more user friendly and consistent with cirq which general tends towards longer, more descriptive names.

Reply:
Just the function right?